HLTrigger/Configuration/python/customizeHLTforCMSSW.py

```diff
@@ -53,11 +53,21 @@ def customiseFor25811(process):
             prod.MissCalibrate = cms.bool(prod.MissCalibrate.value())
     return process
 
+# type of parameter changed
+def customiseFor24501(process):
+    for producer in producers_by_type(process, "PFRecoTauDiscriminationByIsolation"):
+        if hasattr(producer, "deltaBetaPUTrackPtCutOverride"):
```

Comment:
This customization behaves correctly until the modified PFRecoTauDiscriminationByIsolation is parsed in ConfDB. After that, all configs will have a bool deltaBetaPUTrackPtCutOverride parameter, which would be used to replace a double parameter by this customization. Therefore, as soon as the new PFRecoTauDiscriminationByIsolation is parsed this customization must be removed, otherwise it will give rise to wrong configs.

Maybe a safer way to customize could be
```python
        if hasattr(producer, "deltaBetaPUTrackPtCutOverride") and isinstance(producer.deltaBetaPUTrackPtCutOverride, bool):
```

But perhaps this is just an overprotection...
@Martin-Grunewald what do you think?

Reply:
Good point!

Reply:
Indeed, I'll commit it in a second...

Reply:
I add a 'not' before isinstance because everything is fine if it already is bool. Does isinstance recognize cms.bool as bool?

Reply:
cms.bool is not a python bool instance.

Reply:
Seems not to work, I write `if hasattr(producer, "deltaBetaPUTrackPtCutOverride") and not isinstance(producer.deltaBetaPUTrackPtCutOverride, cms.bool):`